packages/buckets/src/api/index.ts

```diff
@@ -52,9 +53,32 @@ import { File, normaliseInput } from './normalize'
 
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const block = require('it-block')
-
 const logger = log.getLogger('buckets-api')
 
+/**
```

Comment:
Pulled this out into its own interface, before it was just the progress thing, now we have root and signal.

Reply:
could you add this to sidebar.js?

Reply:
Can you actually set that as a "request change" thing in your review, I'm about to walk the dog and I don't wanna forget.